main.go

```diff
@@ -125,116 +127,157 @@ func main() {
 		}
 	}
 
-	if err = (&controllers.QueueReconciler{
+	queueRecorder := mgr.GetEventRecorderFor(controllers.QueueControllerName)
+	if err = (&controllers.TopologyReconciler{
 		Client:                  mgr.GetClient(),
+		Type:                    &topology.Queue{},
 		Log:                     ctrl.Log.WithName(controllers.QueueControllerName),
 		Scheme:                  mgr.GetScheme(),
 		Recorder:                mgr.GetEventRecorderFor(controllers.QueueControllerName),
 		RabbitmqClientFactory:   rabbitmqclient.RabbitholeClientFactory,
 		KubernetesClusterDomain: clusterDomain,
+		ReconcileFunc:           &controllers.QueueReconciler{Recorder: queueRecorder, Client: mgr.GetClient()},
```

Comment:
This TopologyReconciler object contains essentially an embedded version of itself, since the QueueReconciler is not too different from the TopologyReconciler object. We're even instantiating it with the same Recorder and Client as the parent type, and that duplication implies to me that there's some bleed between the two in responsibility.

Looking at the old reconcilers, I'm not actually convinced we need to keep them as objects anymore - the two methods DeclareFunc and DeleteFunc very rarely access properties on their respective objects anymore, and so could be functions.

I think with that in mind, the DeclareFunc and DeleteFunc could be converted from methods into functions, and then this property of the TopologyReconciler can literally be a function signature, i.e.

```go
// TopologyReconciler reconciles any topology rabbitmq objects
type TopologyReconciler struct {
	client.Client
	Type                    client.Object
	WatchTypes              []client.Object
	Log                     logr.Logger
	Scheme                  *runtime.Scheme
	Recorder                record.EventRecorder
	RabbitmqClientFactory   rabbitmqclient.Factory
	KubernetesClusterDomain string
        DeleteFunc              func(ctx context.Context, client rabbitmqclient.Client, resource topology.TopologyResource) error
	DeleteFunc              func(ctx context.Context, client rabbitmqclient.Client, resource topology.TopologyResource) error
}
```

```go
TopologyReconciler{
  Client:                  mgr.GetClient(),
  Type:                    &topology.Queue{},
  Log:                     ctrl.Log.WithName(controllers.QueueControllerName),
  Scheme:                  mgr.GetScheme(),
  Recorder:                mgr.GetEventRecorderFor(controllers.QueueControllerName),
  RabbitmqClientFactory:   rabbitmqclient.RabbitholeClientFactory,
  KubernetesClusterDomain: clusterDomain,
  DeclareFunc:             &controllers.QueueDeclare,
  DeleteFunc:              &controllers.QueueDelete,
}
```

What do you think?